### PR TITLE
Add a new release of mlmpfr.4.1.0 with bug fixes

### PR DESCRIPTION
**Pull request by thvnx**

### CODE REVIEW

File: packages/mlmpfr/mlmpfr.4.1.0+bugfix1/opam [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  ["./mlmpfr_compatibility_test"]
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "ocaml" {>= "4.04"}
+  "dune" {>= "1.11.0"}
+]
+x-ci-accept-failures: ["debian-unstable"]
+depexts: [
```
Comment (mseri) [inline packages/mlmpfr/mlmpfr.4.1.0+bugfix1/opam:19]: Please remove the depexts and use `conf-mpfr` instead.
(I will do the same for the old packages that need it once this PR is merged)

File: packages/mlmpfr/mlmpfr.4.1.0+bugfix1/opam [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+dev-repo:     "git+https://github.com/thvnx/mlmpfr.git"
+build: [
+  ["cc" "mlmpfr_compatibility_test.c" "-lmpfr" "-o" "mlmpfr_compatibility_test"]
+  ["./mlmpfr_compatibility_test"]
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "ocaml" {>= "4.04"}
+  "dune" {>= "1.11.0"}
```
Comment (mseri) [inline packages/mlmpfr/mlmpfr.4.1.0+bugfix1/opam:16]:
```suggestion
  "dune" {>= "2.9"}
```
According to the dune-project file

**Comment by mseri** [comment]
`why3 1.4.1` requires an upper bound on mlpmfr:
```
# Error: The implementation src/util/mlmpfr_wrapper.ml
#        does not match the interface src/util/mlmpfr_wrapper.cmi: 
#        Values do not match:
#          val get_formatted_str :
#            ?rnd:mpfr_rnd_t ->
#            ?base:int -> ?size:int -> ?ktz:bool -> mpfr_float -> string
#        is not included in
#          val get_formatted_str :
#            ?rnd:mpfr_rnd_t -> ?base:int -> ?size:int -> mpfr_float -> string
#        The type
#          ?rnd:mpfr_rnd_t ->
#          ?base:int -> ?size:int -> ?ktz:bool -> mpfr_float -> string
#        is not compatible with the type
#          ?rnd:mpfr_rnd_t -> ?base:int -> ?size:int -> mpfr_float -> string
#        Type ?ktz:bool -> mpfr_float -> string is not compatible with type
#          mpfr_float -> string 
#        File "src/util/mlmpfr_wrapper.mli", line 66, characters 0-89:
#          Expected declaration
#        File "src/mpfr.mli", line 197, characters 0-102: Actual declaration
```

File: packages/mlmpfr/mlmpfr.4.1.0+bugfix1/opam (hunk 1 above)
Comment (thvnx) [reply packages/mlmpfr/mlmpfr.4.1.0+bugfix1/opam]: Thanks!

**Comment by thvnx** [comment]
How should I run `why3` to reproduce the error?

**Comment by thvnx** [comment]
> How should I run `why3` to reproduce the error?

Never mind, CI results are back! I'll have a look at them.

**Comment by thvnx** [comment]
OK, I guess that the problem comes from a dependency issue between `why3` and `mlmpfr`. `why3` depends on `mlmpfr` but looks outdated since the signature of `get_formatted_str` has changed since `mlmpfr.4.1.0`. `why3` should be updated but I can't create issue on their repo, see: https://gitlab.inria.fr/why3/why3/-/blob/master/src/util/mlmpfr_wrapper.mli.

@mseri, how should we proceed here, is it possible to merge this and fix `why3` later, or should be the other way around?

**Comment by mseri** [comment]
No need to fix why 3. It is enough to add an upper bound to mlmpfr in the opam file of why3 in the opam repository. I am happy to do that but I will not be able to do it before tonight or tomorrow

**Comment by thvnx** [comment]
> No need to fix why 3. It is enough to add an upper bound to mlmpfr in the opam file of why3 in the opam repository. I am happy to do that but I will not be able to do it before tonight or tomorrow

Oh! Got it, will do it thanks!

**Comment by mseri** [comment]
Thanks!

**Comment by silene** [comment]
Why was Why3 1.5.0 marked as incompatible with mlmpfr 4.1.0, while the revdeps job succeeded?

**Comment by silene** [comment]
Why was Why3 1.5.0 marked as incompatible with mlmpfr 4.1.0, while the revdeps job succeeded?

**Comment by mseri** [comment]
The revdeps job had failed in the original version of this PR with:
```
# Error: The implementation src/util/mlmpfr_wrapper.ml
#        does not match the interface src/util/mlmpfr_wrapper.cmi: 
#        Values do not match:
#          val get_formatted_str :
#            ?rnd:mpfr_rnd_t ->
#            ?base:int -> ?size:int -> ?ktz:bool -> mpfr_float -> string
#        is not included in
#          val get_formatted_str :
#            ?rnd:mpfr_rnd_t -> ?base:int -> ?size:int -> mpfr_float -> string
#        The type
#          ?rnd:mpfr_rnd_t ->
#          ?base:int -> ?size:int -> ?ktz:bool -> mpfr_float -> string
#        is not compatible with the type
#          ?rnd:mpfr_rnd_t -> ?base:int -> ?size:int -> mpfr_float -> string
#        Type ?ktz:bool -> mpfr_float -> string is not compatible with type
#          mpfr_float -> string 
#        File "src/util/mlmpfr_wrapper.mli", line 66, characters 0-89:
#          Expected declaration
#        File "src/mpfr.mli", line 197, characters 0-102: Actual declaration
```

**Comment by silene** [comment]
That is for Why3 1.4.1, not for Why3 1.5.0, or am I missing something?

**Comment by mseri** [comment]
Oh, sorry, you are completely right, that was 1.4.1. The 1.5.0 was there by mistake. I'll remove the bound

**Comment by mseri** [comment]
I just realized that the bound was added in an incorrect, place, opam < 2.2 does ignore it. So it is a change with no real effect. (I am still going to revert it soon)

**Comment by thvnx** [comment]
> Why was Why3 1.5.0 marked as incompatible with mlmpfr 4.1.0, while the revdeps job succeeded?

my mistake, sorry about that!

**Pull request merged**